core/state_transition.go

```diff
@@ -378,16 +378,12 @@ func (st *StateTransition) debitFee(from common.Address, feeCurrency *common.Add
 		st.state.SubBalance(from, effectiveFee)
 		return nil
 	} else {
-		var currencyFee *big.Int
-		if st.msg.MaxFeeInFeeCurrency() == nil {
-			// Normal feeCurrency tx
-			currencyFee = effectiveFee
+		if st.msg.MaxFeeInFeeCurrency() != nil {
+			st.erc20FeeDebited = denominatedCurrencyRate.FromBase(effectiveFee)
 		} else {
-			// Celo denominated tx
-			currencyFee = feeCurrencyRate.FromBase(effectiveFee)
+			st.erc20FeeDebited = effectiveFee
 		}
-		st.erc20FeeDebited = currencyFee
-		return erc20gas.DebitFees(st.evm, from, currencyFee, feeCurrency)
```

Comment:
This block also seems to be functionally unchanged

Reply:
Just removing an unnecessary local variable.